File: prozesse_und_threads/vorgabe/01_fork.py
#!/usr/bin/env python3

import sys
import os

if __name__ == "__main__":
    limit = 100
    pid = os.fork()
    if pid < 0:
        print("Fork failed")
        sys.exit(-1)
    elif pid == 0:
        print("begin") # TODO Info: Child oder Parent?
        for i in range(0, limit):
            pass # TODO Ausgabe: Eltern-oder Kindprozess mit counter
        print("end") # TODO Info: Child oder Parent?
        os._exit(0)
    else:
        print("begin") # TODO Info: Child oder Parent?
        for j in range(0, limit):
            pass # TODO Ausgabe: Eltern-oder Kindprozess mit counter
        print("end") # TODO Info: Child oder Parent?
        sys.exit(0)
        

File: prozesse_und_threads/vorgabe/01_fork.cpp
#include <unistd.h>
#include <iostream>

using namespace std;

int main()
{
    int limit = 100;
    int counter = 0;
    pid_t pid = fork();

    if (pid == 0)
    {
        cout << "begin" << endl; // TODO: child or parent?
        for (int i = 0; i < limit; i++)
        {
            // TODO: child or parent?
        }
        cout << "end" << endl; // TODO: child or parent?
    }
    else if (pid > 0)
    {
        cout << "begin parent" << endl; // TODO: child or parent?
        for (int j = 0; j < limit; j++)
        {
            // TODO: child or parent?
        }
        cout << "end" << endl; // TODO: child or parent?
    }
    else
    {
        cout << "Fork failed!" << endl;
        return -1;
    }
    return 0;
}